src/Collections/SavedTokens.php

```diff
@@ -64,20 +64,23 @@ public function create($token_string)
      */
     public function deleteAll()
     {
-        foreach ($this->getMembers() as $token) {
+        foreach ($this->all() as $token) {
             $token->delete();
         }
     }
 
     /**
      * @inheritdoc
      */
-    protected function getCollectionData($options = [])
+    public function getData()
     {
-        $tokens = [];
-        foreach ($this->getDataStore()->keys() as $key) {
-            $tokens[] = (object)$this->getDataStore()->get($key);
+        if (empty(parent::getData())) {
```

Comment:
This is triggering my php alarms, but I guess it's no longer a problem since we don't support older versions of php. Old instincts die hard.

Reply:
At first I was using `$this->data`, but then I fought it: "Do you want to change every instance of it if you start doing something else with the data?" While I couldn't decide *what* might change, I figured I ought to encapsulate against the future anyway.